File: backchannel-authenticator-archetype/src/main/resources/archetype-resources/src/main/java/descriptor/__pluginName__BackchannelAuthenticatorPluginDescriptor.java
package ${package}.descriptor;

import ${package}.authentication.${pluginName}BackchannelAuthenticatorRequestHandler;
import ${package}.config.${pluginName}BackchannelAuthenticatorPluginConfig;
import se.curity.identityserver.sdk.Nullable;
import se.curity.identityserver.sdk.authentication.BackchannelAuthenticationHandler;
import se.curity.identityserver.sdk.plugin.descriptor.AuthenticatorPluginDescriptor;
import se.curity.identityserver.sdk.plugin.descriptor.BackchannelAuthenticatorPluginDescriptor;

public final class ${pluginName}BackchannelAuthenticatorPluginDescriptor implements BackchannelAuthenticatorPluginDescriptor<${pluginName}BackchannelAuthenticatorPluginConfig>
{
    @Override
    public String getPluginImplementationType()
    {
        return "${pluginName.toLowerCase()}";
    }

    @Override
    public Class<? extends ${pluginName}BackchannelAuthenticatorPluginConfig> getConfigurationType()
    {
        return ${pluginName}BackchannelAuthenticatorPluginConfig.class;
    }

    @Override
    public Class<? extends BackchannelAuthenticationHandler> getBackchannelAuthenticationHandlerType()
    {
        return ${pluginName}BackchannelAuthenticatorRequestHandler.class;
    }

    @Override
    public @Nullable Class<? extends AuthenticatorPluginDescriptor<?>> getFrontchannelPluginDescriptorReference()
    {
        return ${pluginName}AuthenticatorPluginDescriptor.class;
    }
}

File: kotlin-backchannel-authenticator-archetype/src/main/resources/archetype-resources/src/main/kotlin/descriptor/__pluginName__BackchannelAuthenticatorPluginDescriptor.kt
package ${package}.descriptor

import ${package}.authentication.${pluginName}BackchannelAuthenticatorRequestHandler
import ${package}.config.${pluginName}BackchannelAuthenticatorPluginConfig
import se.curity.identityserver.sdk.Nullable
import se.curity.identityserver.sdk.authentication.BackchannelAuthenticationHandler
import se.curity.identityserver.sdk.plugin.descriptor.AuthenticatorPluginDescriptor
import se.curity.identityserver.sdk.plugin.descriptor.BackchannelAuthenticatorPluginDescriptor


class ${pluginName}BackchannelAuthenticatorPluginDescriptor : BackchannelAuthenticatorPluginDescriptor<${pluginName}BackchannelAuthenticatorPluginConfig>
{
    override fun getBackchannelAuthenticationHandlerType(): Class<out BackchannelAuthenticationHandler>
    {
        return ${pluginName}BackchannelAuthenticatorRequestHandler::class.java
    }

    override @Nullable fun getFrontchannelPluginDescriptorReference(): Class<out AuthenticatorPluginDescriptor<*>>
    {
        return ${pluginName}AuthenticatorPluginDescriptor::class.java
    }

    override fun getConfigurationType(): Class<out ${pluginName}BackchannelAuthenticatorPluginConfig> =
        ${pluginName}BackchannelAuthenticatorPluginConfig::class.java

    override fun getPluginImplementationType(): String = "${pluginName.toLowerCase()}"
}
